From 4b82a4e19bbd9211ab6171bf0e2311bc40a32fd7 Mon Sep 17 00:00:00 2001
Message-Id: <4b82a4e19bbd9211ab6171bf0e2311bc40a32fd7.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Tue, 27 Jan 2015 15:27:47 +0530
Subject: [PATCH 62/95] MUST_REBASE [VPG]: drivers/video/adf: Generalize
 sequence for DSI-A and DSI-C

For DSI single link on Port A & C, the seq_port value which is
parsed from Sequence Block#53 of VBT has been set to 0 for both ports.
This allows the same xml to be used for stitching sequence into VBT for
both ports.Now, read/write of packets for the DSI single link on Port A
and Port C will based on the DVO port from VBT block 2.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5584
Change-Id: I2ef40a4891e017c2256944cee125647a0b4924d6
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 .../core/common/dsi/panels/intel_dsi_panel_vbt.c   |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
index 11a7142..300f323 100644
--- a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
+++ b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
@@ -253,7 +253,17 @@ static u8 *mipi_exec_send_packet(struct dsi_pipe *dsi_pipe, u8 *data)
 	vc = (byte >> MIPI_VIRTUAL_CHANNEL_SHIFT) & 0x3;
 	seq_port = (byte >> MIPI_PORT_SHIFT) & 0x3;
 
-	port = intel_dsi_seq_port_to_port(seq_port);
+	/*
+	 * For DSI single link on Port A & C, the seq_port value which is
+	 * parsed from Sequence Block#53 of VBT has been set to 0
+	 * Now, read/write of packets for the DSI single link on Port A and
+	 * Port C will based on the DVO port from VBT block 2.
+	 */
+	if (intel_dsi->ports == (1 << PORT_C))
+		port = PORT_C;
+	else
+		port = intel_dsi_seq_port_to_port(seq_port);
+
 	/* LP or HS mode */
 	intel_dsi->hs = mode;
 
-- 
1.7.9.5

